packages/runtime/src/state/actions.ts

```diff
@@ -803,3 +803,10 @@ export function setLocalizedResourceId({
 export function updateAPIClientCache(payload: APIClientCache): UpdateAPIClientCache {
   return { type: ActionTypes.UPDATE_API_CLIENT_CACHE, payload }
 }
+
+export function isKnownAction(action: unknown): action is Action {
+  return (
+    isAction(action) &&
+    Object.hasOwn(ActionTypes, action.type) != null
+  )
+}
```

Comment:
The two most significant breaking changes in this upgrade are:
- [Middleware `action` and `next` are typed as unknown](https://redux-toolkit.js.org/usage/migrating-rtk-2#middleware-type-changed---middleware-action-and-next-are-typed-as-unknown)
- [Reducers are expected to take `UnknownAction`](https://redux-toolkit.js.org/usage/migrating-rtk-2#anyaction-deprecated-in-favour-of-unknownaction)

`isKnownAction` allows us to limit the impact of these, see its usage in `toolkit.ts` below: https://github.com/makeswift/makeswift/pull/1090/files#diff-2d819ffea223200758ded5b0f0c1e50bb87cb0a227fb7428c93e211d72092ca0